app/code/Magento/Catalog/view/adminhtml/web/catalog/product/attribute/unique-validate.js

```diff
@@ -25,21 +31,27 @@ define([
                             .closest('table')
                             .find('.' + _config.uniqueClass + ':visible'),
                         valuesHash = {},
-                        isValid = true;
+                        isValid = true,
+                        duplicates = [];
 
                     inputs.each(function (el) {
                         var inputValue = inputs[el].value;
 
                         if (typeof valuesHash[inputValue] !== 'undefined') {
                             isValid = false;
+                            duplicates.push(inputValue);
                         }
                         valuesHash[inputValue] = el;
                     });
 
+                    if (!isValid) {
+                        msg = _config.message.replace('%1', duplicates.join(', '));
+                    }
+
                     return isValid;
                 },
 
-                _config.message
+                messager
```

Comment:
@omiroshnichenko No, you will be in a javascript closure - don't know the exact javascript scope name but I think it's a closure - . You have to use custom function to display the msg variable. I tried multiple times like code below and it's not working. 

```
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

define([
    'jquery',
    'mage/backend/validation'
], function (jQuery) {
    'use strict';

    return function (config) {
        var msg = '',
            _config = jQuery.extend({
                element: null,
                message: '',
                uniqueClass: 'required-unique'
            }, config);

        if (typeof _config.element === 'string') {
            jQuery.validator.addMethod(
                _config.element,

                function (value, element) {
                    var inputs = jQuery(element)
                            .closest('table')
                            .find('.' + _config.uniqueClass + ':visible'),
                        valuesHash = {},
                        isValid = true,
                        duplicates = [];

                    inputs.each(function (el) {
                        var inputValue = inputs[el].value;

                        if (typeof valuesHash[inputValue] !== 'undefined') {
                            isValid = false;
                            duplicates.push(inputValue);
                        }
                        valuesHash[inputValue] = el;
                    });

                    if (!isValid) {
                        msg = _config.message.replace('%1', duplicates.join(', '));
                    }

                    return isValid;
                },

                msg
            );
        }
    };
});
```